framework/src/modules/dpos_v2/method.ts

```diff
@@ -74,12 +79,53 @@ export class DPoSMethod extends BaseMethod {
 	}
 
 	public async updateSharedRewards(
-		_methodContext: MethodContext,
-		_generatorAddress: Buffer,
-		_tokenID: Buffer,
-		_reward: bigint,
+		context: MethodContext,
+		generatorAddress: Buffer,
+		tokenID: Buffer,
+		reward: bigint,
 	): Promise<void> {
-		// TODO: Implement #7715
+		const delegateStore = this.stores.get(DelegateStore);
+		const delegate = await delegateStore.get(context, generatorAddress);
+		if (delegate.totalVotesReceived === BigInt(0)) {
+			return;
+		}
+
+		const { q96 } = math;
+		const rewardQ = q96(reward);
+		const commissionQ = q96(BigInt(delegate.commission));
+		const rewardFractionQ = q96(BigInt(1)).sub(commissionQ.div(q96(BigInt(10000))));
+		const selfVotesQ = q96(delegate.selfVotes);
+		const totalVotesQ = q96(delegate.totalVotesReceived);
+
+		let hasItem = false;
+		for (const item of delegate.sharingCoefficients) {
+			if (item.tokenID.equals(tokenID)) {
+				hasItem = true;
+				break;
+			}
+		}
+		if (!hasItem) {
+			delegate.sharingCoefficients.push({ tokenID, coefficient: q96(BigInt(0)).toBuffer() });
+		}
```

Comment:
maybe simpler to do
```
const matchingCoefficientIndex = delegate.sharingCoefficients.findIndex(coefficient => coefficient.tokenID.equal(tokenID));
const index = matchingCoefficientIndex > -1 ? matchingCoefficientIndex : delegate.sharingCoefficients.length;
if (index < 0) {
  delegate.sharingCoefficients[index] = { tokenID, coefficient: q96(BigInt(0)).toBuffer() }
}
...use index for oldSharing coefficient and update
sort by tokenID and save
```

Reply:
in this way, no need to loop multiple times

Reply:
ah sorry, yea that's not correct.
it should be `if (matchingCoefficientIndex < 0)`